### PR TITLE
feat: Scene-Handler Refactor (FM-300)

### PR DESCRIPTION
**Pull request by bernhardccodev**
# Changes
- Moved Loading Scene logics and handlings from Scene-Handler to loading-scene.ts.
- Hooked up loading-scenes.ts with game state service.
- Updated each scenes to properly hooked up the pause event and did some code clean up.
- Refactor how scene handler handles the switching of scenes and disposing of previous scenes.
- Code clean up with game state service and created new DAO for loading-scene.ts.

# How to test
- Need to play and visit every scenes

Ref: [FM-300](https://curiouslearning.atlassian.net/browse/FM-300)

[FM-300]: https://curiouslearning.atlassian.net/browse/FM-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/sceneHandler/index.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
 import { DataModal, GameScore } from "@data";
 import { Debugger } from "@common";
 import {
+  SCENE_NAME_LOADING,
```
Comment (miguelccodev) [inline src/sceneHandler/index.ts:11]: Add todo to refactor later.

From a quick scan, it looks like the `loading-scene` is really a transition between scenes.
We can reactor this to a transition scene in the scene-handler that would show when changing scenes.

**Review by miguelccodev — COMMENTED** [review]
Minor comments.

File: src/sceneHandler/index.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {
 import { DataModal, GameScore } from "@data";
```
[the remark this answers is not in the document]
Comment (bernhardccodev) [reply src/sceneHandler/index.ts]: File has been renamed to scene-handler.ts.

File: src/sceneHandler/index.ts (hunk 1 above)
Comment (bernhardccodev) [reply src/sceneHandler/index.ts]: The variable SCENE_NAME_LOADING has been renamed to LOADING_TRANSITION.

File: src/sceneHandler/scene-handler.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    SCENE_NAME_LEVEL_SELECT?: LevelSelectionScreen;
+    SCENE_NAME_GAME_PLAY?: GameplayScene;
+    SCENE_NAME_LEVEL_END?: LevelEndScene;
+  };
+  private activeScene: null | StartScene | LevelSelectionScreen | GameplayScene | LevelEndScene;
+  public canvas: HTMLCanvasElement; //Remove and use DAO.
+  public data: DataModal; //Remove and use DAO.
+  public width: number; //Remove and use DAO.
+  public height: number; //Remove and use DAO.
+  public canavsElement: HTMLCanvasElement; //Remove and use DAO.
```
Comment (ashwinnair-chimple) [inline src/sceneHandler/scene-handler.ts:33]: can we correct the typo with a consistent naming "canvasElement"

**Review by ashwinnair-chimple — COMMENTED** [review]
Added a small comment for correcting a typo

File: src/sceneHandler/scene-handler.ts (hunk 3 above)
Comment (bernhardccodev) [reply src/sceneHandler/scene-handler.ts]: Corrected the typo from "canavsElement" to "canvasElement".

**Review by ashwinnair-chimple — APPROVED** [review]
Approved

**Review by miguelccodev — APPROVED** [review]
approved

**Pull request merged**